### ISSUE TITLE
Modules and Plugins with Known Issues Doc Update 

### ISSUE THREAD

**Issue by jonathanstegall** [pantheon-systems/documentation#3684]
Re: [Modules and Plugins with Known Issues](https://pantheon.io/docs/modules-plugins-known-issues/)

Priority (Low‚ Medium‚ High): medium

## Issue Description: I maintain a WordPress plugin that I've heard does not work on Pantheon because it is unable to push its required dependencies from dev to production environments. It is not listed on the page here, but I'd like to have it work in your environment anyway if I can.

This is the plugin: https://github.com/MinnPost/object-sync-for-salesforce

The issue, as far as I can tell, is that it uses composer to manage some of its dependencies. When a user installs the plugin from the WordPress plugin directory (https://wordpress.org/plugins/object-sync-for-salesforce/), they get those dependencies because the WordPress directory has them.

But pushing them from a dev environment to a production environment does not send those dependencies.

Are you able to help me resolve this?

**Comment by sarahg** [comment]
@jonathanstegall Thanks for reporting the issue! 

I can take a look at this on the platform and let you know where things go missing. If I had to guess, I'd suspect maybe a gitignore file bundled in with dependencies somewhere, or maybe something coming through as a git submodule, is preventing files from being committed to the main Pantheon site repo (and thus being deployable). Usually that's fixable with some modifications to composer.json.

Anyways, I'll let you know what I find!

**Comment by sarahg** [comment]
@jonathanstegall Just gave this a test run. When you download the plugin via the WP admin UI, commit the files on Dev, and then deploy to another environment, you run into this fatal error:

`Warning: include(/srv/bindings/2edd2b50dc444117a87d5e8bd1f63b32/code/wp-content/plugins/object-sync-for-salesforce/vendor/composer/../pippinsplugins/wp-logging/WP_Logging.php): failed to open stream: No such file or directory in /srv/bindings/2edd2b50dc444117a87d5e8bd1f63b32/code/wp-content/plugins/object-sync-for-salesforce/vendor/composer/ClassLoader.php on line 444 Warning: include(): Failed opening '/srv/bindings/2edd2b50dc444117a87d5e8bd1f63b32/code/wp-content/plugins/object-sync-for-salesforce/vendor/composer/../pippinsplugins/wp-logging/WP_Logging.php' for inclusion (include_path='.:/usr/share/pear:/usr/share/php') in /srv/bindings/2edd2b50dc444117a87d5e8bd1f63b32/code/wp-content/plugins/object-sync-for-salesforce/vendor/composer/ClassLoader.php on line 444 Fatal error: Class 'WP_Logging' not found in /srv/bindings/2edd2b50dc444117a87d5e8bd1f63b32/code/wp-content/plugins/object-sync-for-salesforce/classes/logging.php on line 15`

**Comment by jonathanstegall** [comment]
@sarahg that's right. The library - WP_Logging.php - comes from composer. I think I'm asking what the practice is to get the composer install to run, since composer installed libraries are not supposed to be in a git repository. For example, when we deploy the plugin to the WordPress repository, we have a script that runs `composer install` right before it pushes to WordPress, and it uses the composer.json to install the libraries.

I'm assuming there's something that we are supposed to do for Pantheon users to get this to occur as well? I'm reluctant to add things to the plugin's repository that are just for Pantheon, if that makes sense

**Comment by sarahg** [comment]
@jonathanstegall Thanks for that info and apologies for the slow reply! We can't run `composer install` directly on our application servers, so when a Composer dependency is required for a plugin, people need to run the install/update locally, or on a CI service, then commit/push all the files back up to Pantheon. 

I think at this point the best workaround will be to note that plugin users need to remove the gitignore files within the plugin. Totally understood on not wanting to add Pantheon-specific code to the plugin itself.

I've requested an internal technical review on the PR I just sent in for this, so we'll get another opinion before getting this up in our docs.
